app/components/graphs/ResaleValueLineChart.tsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import { LineChart, Line, CartesianGrid, XAxis, YAxis, Label, TooltipProps } from 'recharts';
+import { Card, CardContent, CardHeader } from "@/components/ui/card";
+import {
+  ChartConfig,
+  ChartContainer,
+  ChartTooltip,
+} from "@/components/ui/chart";
+import { formatValue } from "@/app/lib/format";
+
+type CustomTooltipProps = TooltipProps<number, string> & {
+  payload?: Array<{
+    name: keyof Omit<DataPoint, 'year'>;
+    value: number;
+    stroke: string;
+  }>;
+};
+
+const chartConfig = {
+  noMaintenance: {
+    label: "No maintenance",
+    color: "rgb(var(--fairhold-land-color-rgb))", 
+  },
+  lowMaintenance: {
+    label: "Low maintenance",
+    color: "rgb(var(--fairhold-land-color-rgb))",
+  },
+  mediumMaintenance: {
+    label: "Medium maintenance",
+    color: "rgb(var(--fairhold-land-color-rgb))",
+  },
+  highMaintenance: {
+    label: "High maintenance",
+    color: "rgb(var(--fairhold-land-color-rgb))",
+  },
+} satisfies ChartConfig;
+
+export interface DataPoint {
+  year: number;
+  noMaintenance: number;
+  lowMaintenance: number;
+  mediumMaintenance: number;
+  highMaintenance: number;
+}
+
+interface ResaleValueLineChartProps {
+  data: DataPoint[];
+  selectedMaintenance: 'noMaintenance' | 'lowMaintenance' | 'mediumMaintenance' | 'highMaintenance';
+  maxY: number;
+}
+
+const ResaleValueLineChart: React.FC<ResaleValueLineChartProps> = ({
+  data,
+  selectedMaintenance,
+  maxY
+}) => {
+  const renderLine = (dataKey: keyof Omit<DataPoint, 'year'>) => (
+    <Line
+      type="monotone"
+      dataKey={dataKey}
+      stroke={`var(--color-${dataKey})`}
+      strokeWidth={2}
+      strokeDasharray={dataKey === selectedMaintenance ? "0" : "5 5"}
+      dot={false}
+    />
+  );
+  const CustomTooltip = ({ active, payload, label }: CustomTooltipProps) => {
+    if (active && payload && payload.length) {
```

Comment:
nit: A guard clause (reversing the condition, returning `null`) would be clearer and easier to follow as we'd contain the logic in one place and lose a level of indentation.